src/ledger/GeneralizedLedgerEntry.h

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#pragma once
+
+#include "xdr/Stellar-ledger-entries.h"
+#include "xdr/Stellar-ledger.h"
+
+namespace stellar
+{
+
+enum class GeneralizedLedgerEntryType
+{
+    LEDGER_ENTRY
+};
+
+class GeneralizedLedgerKey
+{
+  private:
+    GeneralizedLedgerEntryType mType;
+    union {
+        LedgerKey mLedgerKey;
+    };
+
+    void assign(GeneralizedLedgerKey const& glk);
```

Comment:
Could do this, or use pointers and avoid dealing with unions for now? (we'll move to C++ 17 shortly and can use variants).

I know this might be faster, but custom new/delete operators tends to trip asan etc and come with some many warning labels...